sw/device/tests/sleep_pwm_pulses_test.c

```diff
@@ -0,0 +1,205 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_aon_timer.h"
+#include "sw/device/lib/dif/dif_pinmux.h"
+#include "sw/device/lib/dif/dif_pwm.h"
+#include "sw/device/lib/dif/dif_pwrmgr.h"
+#include "sw/device/lib/dif/dif_rstmgr.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/aon_timer_testutils.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/pwrmgr_testutils.h"
+#include "sw/device/lib/testing/rstmgr_testutils.h"
+#include "sw/device/lib/testing/test_framework/ottf.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "pwm_regs.h"
+
+/**
+ * SLEEP PWM PULSES test
+ *
+ * This test configure 6 pwm channels with a fixed duty cycle
+ * then kicks power manager sleep mode to see pwm pluses are
+ * not affected by power down event.
+ * pwm out --> pinmux setup is chosen arbitrary as below
+ *      pwmout[0] -> IOB10
+ *      pwmout[1] -> IOB11
+ *      pwmout[2] -> IOB12
+ *      pwmout[3] -> IOc10
+ *      pwmout[4] -> IOc11
+ *      pwmout[5] -> IOc12
+ *
+ * Since purpose of this test is to check pwm -> pinmux
+ * connectivity and pulse integrity under sleep event,
+ * Fixed pwm configuration, mode and 0 phase delay are chosen.
+ */
+
+static const dif_pinmux_index_t kPinmuxOutsel[PWM_PARAM_N_OUTPUTS] = {
+    kTopEarlgreyPinmuxOutselPwmAonPwm0, kTopEarlgreyPinmuxOutselPwmAonPwm1,
+    kTopEarlgreyPinmuxOutselPwmAonPwm2, kTopEarlgreyPinmuxOutselPwmAonPwm3,
+    kTopEarlgreyPinmuxOutselPwmAonPwm4, kTopEarlgreyPinmuxOutselPwmAonPwm5,
+};
+
+static const dif_pinmux_index_t kPinmuxMioOut[PWM_PARAM_N_OUTPUTS] = {
+    kTopEarlgreyPinmuxMioOutIob10, kTopEarlgreyPinmuxMioOutIob11,
+    kTopEarlgreyPinmuxMioOutIob12, kTopEarlgreyPinmuxMioOutIoc10,
+    kTopEarlgreyPinmuxMioOutIoc11, kTopEarlgreyPinmuxMioOutIoc12,
+};
+
+static const dif_pwm_channel_t kPwmChannel[PWM_PARAM_N_OUTPUTS] = {
+    kDifPwmChannel0, kDifPwmChannel1, kDifPwmChannel2,
+    kDifPwmChannel3, kDifPwmChannel4, kDifPwmChannel5,
+};
+
+// Duty cycle in the unit of beat
+// These are random numbers betwen [1,beats_per_pulse_cycle)
+// make 'static volatile' to overwrite from sv test
+static volatile const uint16_t kPwmDutycycle[PWM_PARAM_N_OUTPUTS] = {
```

Comment:
Add a comment here stating that the DV test sequence `<path-to-vseq>` may overwrite these values via backdoor.